src/wh_server.c

```diff
@@ -3,40 +3,93 @@
+
+/* Server Components */
 #include "wolfhsm/wh_comm.h"
+#include "wolfhsm/wh_nvm.h"
 
+/* Message definitions */
 #include "wolfhsm/wh_message.h"
 #include "wolfhsm/wh_message_comm.h"
+#include "wolfhsm/wh_message_nvm.h"
 #include "wolfhsm/wh_packet.h"
+
+/* Server API's */
 #include "wolfhsm/wh_server.h"
 #include "wolfhsm/wh_server_crypto.h"
 #include "wolfhsm/wh_server_she.h"
+#include "wolfhsm/wh_server_internal.h"
+
+/** Forward declarations. */
+/* TODO: Move these out to separate C files */
+static int _wh_Server_HandleCommRequest(whServerContext* server,
+        uint16_t magic, uint16_t action, uint16_t seq,
+        uint16_t req_size, const void* req_packet,
+        uint16_t *out_resp_size, void* resp_packet);
+static int _wh_Server_HandleKeyRequest(whServerContext* server,
+        uint16_t magic, uint16_t action, uint16_t seq,
+        uint16_t req_size, const void* req_packet,
+        uint16_t *out_resp_size, void* resp_packet);
+static int _wh_Server_HandlePkcs11Request(whServerContext* server,
+        uint16_t magic, uint16_t action, uint16_t seq,
+        uint16_t req_size, const void* req_packet,
+        uint16_t *out_resp_size, void* resp_packet);
+#ifdef WOLFHSM_SHE_EXTENSION
+static int _wh_Server_HandleSheRequest(whServerContext* server,
+        uint16_t magic, uint16_t action, uint16_t seq,
+        uint16_t req_size, const void* req_packet,
+        uint16_t *out_resp_size, void* resp_packet);
+#endif
+static int _wh_Server_HandleCustomRequest(whServerContext* server,
+        uint16_t magic, uint16_t action, uint16_t seq,
+        uint16_t req_size, const void* req_packet,
+        uint16_t *out_resp_size, void* resp_packet);
+
 
 int wh_Server_Init(whServerContext* server, whServerConfig* config)
 {
     int rc = 0;
-    if (server == NULL) {
+    if ((server == NULL) || (config == NULL)) {
         return WH_ERROR_BADARGS;
     }
 
     memset(server, 0, sizeof(*server));
     if (
-/*            ((rc = wh_Nvm_Init(server->nvm_device, config->nvm_device)) == 0) && */
-        ((rc = wh_CommServer_Init(server->comm, config->comm)) == 0) &&
         ((rc = wolfCrypt_Init()) == 0) &&
         ((rc = wc_InitRng_ex(server->crypto->rng, NULL, INVALID_DEVID)) == 0) &&
-/*        ((rc = server->nvm->cb->Init(server->nvm, config->nvm)) == 0) && */
         1) {
-        /* All good */
+        rc = wh_Nvm_Init(server->nvm, config->nvm_config);
+        if (rc == 0) {
+            server->nvm->cb = config->nvm_config->cb;
+            server->nvm->context = config->nvm_config->context;
+
+            rc = wh_CommServer_Init(server->comm, config->comm_config);
+            if (rc == 0) {
+                /* All good */
+            }
+        }
     } else {
         wh_Server_Cleanup(server);
     }
-    /* WC_INIT_E, WC_HW_E*/
     return rc;
 }
 
+int wh_Server_Cleanup(whServerContext* server)
+{
+    if (server ==NULL) {
+        return WH_ERROR_BADARGS;
+    }
+
+    (void)wh_CommServer_Cleanup(server->comm);
```

Comment:
should we be ignoring the return codes here? if cleanup fails the client won't know if they tried to init again